apps/studio/components/ui/AIAssistantPanel/AssistantChatForm.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { ChangeEvent, memo, useRef } from 'react'
+import { Button, CodeBlock, ExpandingTextArea } from 'ui'
+import { cn } from 'ui/src/lib/utils'
+import { HoverCard_Shadcn_, HoverCardTrigger_Shadcn_, HoverCardContent_Shadcn_ } from 'ui'
+
+export interface FormProps {
+  /* The ref for the textarea, optional. Exposed for the CommandsPopover to attach events. */
+  textAreaRef?: React.RefObject<HTMLTextAreaElement>
+  /* The loading state of the form */
+  loading: boolean
+  /* The disabled state of the form */
+  disabled?: boolean
+  /* The value of the textarea */
+  value?: string
+  /* The function to handle the value change */
+  onValueChange: (value: ChangeEvent<HTMLTextAreaElement>) => void
+  /**
+   * If true, include SQL snippets in the message sent to onSubmit
+   */
+  includeSnippetsInMessage?: boolean
+  /**
+   * The function to handle the form submission
+   */
+  onSubmit: (message: string) => void
+  /* The placeholder of the textarea */
+  placeholder?: string
+  /* SQL snippets to display above the form */
+  sqlSnippets?: string[]
+  /* Function to handle removing a SQL snippet */
+  onRemoveSnippet?: (index: number) => void
+  /* Additional class name for the snippets container */
+  snippetsClassName?: string
+  /* Additional class name for the form wrapper */
+  className?: string
+}
+
+const AssistantChatFormComponent = React.forwardRef<HTMLFormElement, FormProps>(
+  (
+    {
+      loading = false,
+      disabled = false,
+      value = '',
+      textAreaRef,
+      onValueChange,
+      onSubmit,
+      placeholder,
+      sqlSnippets,
+      onRemoveSnippet,
+      snippetsClassName,
+      includeSnippetsInMessage = false,
+      className,
+      ...props
+    },
+    ref
+  ) => {
+    const formRef = useRef<HTMLFormElement>(null)
+    const isMobile = useBreakpoint('md')
+
+    const handleSubmit = (event?: React.FormEvent<HTMLFormElement>) => {
+      if (event) event.preventDefault()
+      if (!value || loading) return
+
+      let finalMessage = value
+      if (includeSnippetsInMessage && sqlSnippets && sqlSnippets.length > 0) {
+        const sqlSnippetsString = sqlSnippets
+          .map((snippet: string) => '```sql\n' + snippet + '\n```')
+          .join('\n')
+        finalMessage = [value, sqlSnippetsString].filter(Boolean).join('\n\n')
+      }
+
+      onSubmit(finalMessage)
+    }
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+      if (event.key === 'Enter' && !event.shiftKey) {
+        event.preventDefault()
+        handleSubmit()
+      }
+    }
+
+    const canSubmit = !loading && !!value
+
+    return (
+      <div className="w-full">
+        <form
+          id="assistant-chat"
+          ref={formRef}
+          {...props}
+          onSubmit={handleSubmit}
+          className={cn('relative overflow-hidden', className)}
+        >
+          {/* Snippet badges row */}
+          {sqlSnippets && sqlSnippets.length > 0 && (
```

Comment:
This `snippet badges row` should be extracted into a component and used where needed. There's no other changes to `AssistantChatFormComponent` when compared to `AssistantChatForm`, right?